chain/chunks/src/lib.rs

```diff
@@ -1293,18 +1292,20 @@ mod test {
         let network_adapter = Arc::new(MockNetworkAdapter::default());
         let mut shards_manager =
             ShardsManager::new(Some("test".to_string()), runtime_adapter, network_adapter.clone());
-        shards_manager.requested_partial_encoded_chunks.insert(
-            ChunkHash(hash(&[1])),
-            ChunkRequestInfo {
-                height: 0,
-                parent_hash: Default::default(),
-                shard_id: 0,
-                added: Instant::now(),
-                last_requested: Instant::now(),
-            },
-        );
-        std::thread::sleep(Duration::from_millis(200));
```

Comment:
I suspect that this sleep and the separation of pushing / requesting was here for a reason. For the first `CHUNK_REQUEST_SWITCH_TO_OTHERS_MS` milliseconds we do not request chunks from anyone but the part owners / chunk producer, and only then we start requesting from other participants (block producers or peers) who track the shard, thus the bug with requesting from ourselves would only repro if we separated pushing the request, and actually requesting.

Though the `CHUNK_REQUEST_SWITCH_TO_OTHERS_MS` is actually larger than 200ms, so this test was not correct to begin with :)

@bowenwang1996 can you confirm I understand the test correctly, and that the sleep duration needs to be increased?

Reply:
I we want to test some particular branch of the code we could still do this without a sleep, say by setting `request_own_parts_from_others = true` in the `request_partial_encoded_chunk` call.

Explicit sleep calls in tests are a bit of a red flag to me in general because it means the code might not be factored in the most testable way. But if we do need to leave the `sleep` in for some reason then it should be relative to the appropriate constant (like `CHUNK_REQUEST_SWITCH_TO_OTHERS_MS`), rather than a "magic number" (like 200).

Reply:
Notably the test passes independent of the value of `request_own_parts_from_others`. I'm not sure if that is intentional or not.

Reply:
@SkidanovAlex the test does not need to sleep for longer than `CHUNK_REQUEST_SWITCH_TO_OTHERS_MS` because there is only one node tracking this shard and in this case it will try to request from that node as long as it sleeps more than `CHUNK_REQUEST_RETRY_MS`, which is 100ms. I agree that it is not ideal to have tests depend on some specific amount of sleep. I think the reason I didn't use `request_partial_encoded_chunk` directly is that I wanted to simulate what happens when a node receives a chunk.

Reply:
If you prefer the version with the sleep then I can revert, but change the `200` to `2 * CHUNK_REQUEST_RETRY_MS`.

Reply:
@birchmd yeah let's do that

Reply:
Sounds good. Done in https://github.com/nearprotocol/nearcore/pull/2669/commits/2761fdd6728c567415ac31f614098685cbcc97da